server/rules-engine/factHandler.js

```diff
@@ -0,0 +1,150 @@
+const jp = require('jsonpath')
+
+const sumallRuleHandler = async function (event, almanac) {
+  const source = await almanac.factValue(event.params.calculation.sourcefact)
+  let totalSum = 0
+  const values = jp.query(source, event.params.calculation.path)
+  totalSum = values.reduce((acc, value) => {
+    acc += value
+    return acc
+  }, totalSum)
+  almanac.addRuntimeFact(event.params.id, totalSum)
+}
+
+const sumallRule = function (fact, priority) {
+  return {
+    conditions: {
+      all: [{
+        fact: fact.id,
+        operator: 'notEqual',
+        value: 0
+      }]
+    },
+    event: { type: 'sumall', params: fact },
+    onSuccess: sumallRuleHandler,
+    priority
+  }
+}
+
+const subtractRuleHandler = async function (event, almanac) {
+  const value1 = await almanac.factValue(event.params.calculation.value1)
+  const value2 = await almanac.factValue(event.params.calculation.value2)
+  almanac.addRuntimeFact(event.params.id, value1 - value2)
+}
+
+const subtractRule = function (fact, priority) {
+  return {
+    conditions: {
+      all: [{
+        fact: fact.id,
+        operator: 'notEqual',
+        value: 0
+      }]
+    },
+    event: { type: 'subtract', params: fact },
+    onSuccess: subtractRuleHandler,
+    priority
+  }
+}
+
+const factHandler = {
+  factsFromRules: function (rules) {
+    const allFacts = rules.reduce((acc, rule) => {
+      if (Object.prototype.hasOwnProperty.call(rule, 'facts')) {
```

Comment:
A great idea, however the linter hates it because it could be overwritten. Seems a daft rule to me, we could possibly review the linter at a later point.